team-aglr/edemame
# cli updates for db schema changes/all db api functionality

how to test updates to cli functionality:

- run a test (from within k6_tests subfolder): `edamame run -p test8.js -n example` (if you don't want to give a specific test name then you can leave out -n example)
- let the test run through to completion
- run: `edamame get --all` <=> should get a table containing info about the test you just ran
- run: `edamame get -n "super test"` <=> should get appropriate message that it couldn't find the test
- run: `edamame get --name example` <=> should get info about the test including the test script contents
- run: `edamame update -c example -n example` <=> should fail gracefully; there is a check to ensure new name doesn't already exist
- run: `edamame update -c example -n "       "` <=> should fail gracefully; there is a check to ensure new name isn't just whitespaces
- run: `edamame update --current example --new "example test"` <=> should successfully update the test name
- run: `edamame delete "example test"` <=> should delete test data
- run another test: `edamame run -p test4.js`
- while test is running, in a different terminal execute: `edamame get --all` <=> you should see table entry with 'starting' or 'running' status depending on when you run execute; note if you time it perfectly and your edamame get request is occurring at the same time as the runTest() function is calling dbApi.updateState(), then one will error out; I'm going to open a separate issue for this, because we don't want runTest to end prematurely, because a different resource is checking the status of a test or getting information about a test (i think one solution might be to update the local port access the dbApi to a different port)
- run: `edamame stop` in a different terminal while the test is still in the running phase; then run `edamame get` and you should see in the table a non-null timestamp under the "end time" column and the "status" should be completed; the terminal that `edamame run` was executed in will never stop, but that's a lower priority issue, b/c the user can just ctrl+c in that terminal (the reason is that the load generators that are polling for complete pods will never find all of them complete, because the test pods will be removed); I couldn't resolve my context missing issue related to the kubernetes/client-node library when executing `edamame stop`, so to test my updates to `edamame stop` I commented out the k8s/client-node library specific lines (which in this commit's proposed commands/stopTest.js file, are lines 5, 8-11, 14-17, 22-24)
